tests/e2e/e2e_test.go

```diff
@@ -1,63 +1,166 @@
 	"testing"
 
 	. "github.com/onsi/ginkgo/v2"
 	. "github.com/onsi/gomega"
 )
 
+const (
+	vectorAddImage   = "nvcr.io/nvidia/k8s/cuda-sample:vectoradd-cuda12.5.0"
+	deviceQueryImage = "nvcr.io/nvidia/k8s/cuda-sample:devicequery-cuda12.5.0"
+	cudaImage        = "nvcr.io/nvidia/cuda:12.8.0-base-ubi8"
+)
+
 // Test context
 var (
 	ctx context.Context
 
 	installCTK bool
 
-	image string
+	ImageRepo string
+	ImageTag  string
 
-	sshKey  string
-	sshUser string
-	host    string
-	sshPort string
-)
+	sshKey      string
+	sshUser     string
+	host        string
+	sshPort     string
+	cwd         string
+	packagePath string
 
-func init() {
-	flag.BoolVar(&installCTK, "install-ctk", false, "Install the NVIDIA Container Toolkit")
-	flag.StringVar(&image, "toolkit-image", "", "Repository of the image to test")
-	flag.StringVar(&sshKey, "ssh-key", "", "SSH key to use for remote login")
-	flag.StringVar(&sshUser, "ssh-user", "", "SSH user to use for remote login")
-	flag.StringVar(&host, "remote-host", "", "Hostname of the remote machine")
-	flag.StringVar(&sshPort, "remote-port", "22", "SSH port to use for remote login")
-}
+	runner Runner
+)
 
 func TestMain(t *testing.T) {
-	suiteName := "NVIDIA Container Toolkit E2E"
+	suiteName := "E2E NVIDIA Container Toolkit"
 
 	RegisterFailHandler(Fail)
+
+	ctx = context.Background()
+	getTestEnv()
+
 	RunSpecs(t,
 		suiteName,
 	)
 }
 
 // BeforeSuite runs before the test suite
 var _ = BeforeSuite(func() {
-	ctx = context.Background()
+	runner = NewRunner(
+		WithHost(host),
+		WithPort(sshPort),
+		WithSshKey(sshKey),
+		WithSshUser(sshUser),
+	)
+
+	if installCTK {
+		installer, err := NewToolkitInstaller(
+			WithRunner(runner),
+			WithImage(ImageRepo+":"+ImageTag),
+			WithTemplate(dockerInstallTemplate),
+		)
+		Expect(err).ToNot(HaveOccurred())
+
+		err = installer.Install()
+		Expect(err).ToNot(HaveOccurred())
+	}
+
+	_, _, err := runner.Run("docker pull ubuntu")
+	Expect(err).ToNot(HaveOccurred())
+
+	_, _, err = runner.Run(fmt.Sprintf("docker pull %s", vectorAddImage))
+	Expect(err).ToNot(HaveOccurred())
+
+	_, _, err = runner.Run(fmt.Sprintf("docker pull %s", deviceQueryImage))
+	Expect(err).ToNot(HaveOccurred())
+	_, _, err = runner.Run(fmt.Sprintf("docker pull %s", cudaImage))
+	Expect(err).ToNot(HaveOccurred())
 })
+
+// getTestEnv gets the test environment variables
+func getTestEnv() {
+	defer GinkgoRecover()
+	var err error
+
+	_, thisFile, _, _ := runtime.Caller(0)
+	packagePath = filepath.Dir(thisFile)
+
+	installCTK = getBoolEnvVar("E2E_INSTALL_CTK", false)
+
+	ImageRepo = os.Getenv("E2E_IMAGE_REPO")
+	Expect(ImageRepo).NotTo(BeEmpty(), "E2E_IMAGE_REPO environment variable must be set")
+
+	ImageTag = os.Getenv("E2E_IMAGE_TAG")
+	Expect(ImageTag).NotTo(BeEmpty(), "E2E_IMAGE_TAG environment variable must be set")
+
+	// TODO (@ArangoGutierrez):
+	// once https://github.com/NVIDIA/nvidia-container-toolkit/pull/602
+	// is merged, remove this
+	ImageTag = fmt.Sprintf("%s-ubuntu20.04", ImageTag)
```

Comment:
This is NOT what I meant by my comment. We should append this suffix in the YAML file and not here.

Reply:
ack